### PR TITLE
formulae: set root_url to GitHub Releases url

### PR DESCRIPTION
**Pull request by dawidd6**
If not testing core tap (and a bunch of other conditions), set the root_url to GitHub releases url.
This url would later be used in `brew pr-upload` to determine what release to create.

### CODE REVIEW

**Review by MikeMcQuaid — APPROVED** [review]
Makes sense to me (minus @Bo98's question I'd like to see figured out). `--ci-upload` should probably be configured to fail with a helpful message given this case, too?

**Comment by dawidd6** [comment]
Yes, `--keep-old` was a bad idea, deleted it.

Also added a check for `--ci-upload` to refuse uploading to GitHub Releases.

File: lib/test_ci_upload.rb [hunk 1]
```diff
@@ -37,6 +37,16 @@ def run!(tap, args:)
         hash.deep_merge(JSON.parse(IO.read(json_file)))
       end
 
+      bottles_hash.each do |_, bottle_hash|
+        root_url = bottle_hash["bottle"]["root_url"]
+        url_match = root_url.match HOMEBREW_RELEASES_URL_REGEX
+        _, _, _, tag = *url_match
+
+        next unless tag
```
Comment (MikeMcQuaid) [inline lib/test_ci_upload.rb:45]:
```suggestion
        next unless root_url.match?(HOMEBREW_RELEASES_URL_REGEX)
```

File: lib/test_ci_upload.rb [hunk 2]
```diff
@@ -37,6 +37,16 @@ def run!(tap, args:)
...
+
+        odie "Cowardly refusing to upload to GitHub Releases, use `brew pr-upload`."
```
Comment (MikeMcQuaid) [inline lib/test_ci_upload.rb:47]:
```suggestion
        odie "Refusing to upload to GitHub Releases, use `brew pr-upload`."
```

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**